packages/tailwindcss/src/utilities.ts

```diff
@@ -1222,21 +1222,51 @@ export function createUtilities(theme: Theme) {
 
   /**
    * @css `rotate`
+   *
+   * `rotate-45` => `rotate: 45deg`
+   * `rotate-45 rotate-x` => `rotate: 1 0 0 45deg`
+   * `rotate-45 rotate-x rotate-y` => `rotate: 1 1 0 45deg`
+   * `rotate-45 rotate-[1_2_3]` => `rotate: 1 2 3 45deg`
    */
-  functionalUtility('rotate', {
-    supportsNegative: true,
-    themeKeys: ['--rotate'],
-    handleBareValue: ({ value }) => `${value}deg`,
-    handle: (value) => [decl('rotate', value)],
+  let rotateProperties = () =>
+    atRoot([
+      property('--tw-rotate', '0deg', '<angle>'),
+      // Vector components for the axis of rotation
+      property('--tw-rotate-x', '0', '<number>'),
+      property('--tw-rotate-y', '0', '<number>'),
+      property('--tw-rotate-z', '0', '<number>'),
+    ])
+  utilities.functional('rotate', (candidate) => {
+    if (!candidate.value) return
+    let value
+    if (candidate.value.kind === 'arbitrary') {
+      value = candidate.value.value
+      let type = candidate.value.dataType ?? inferDataType(value, ['angle', 'vector'])
+      if (type === 'vector') {
+        return [decl('rotate', `${value} var(--tw-rotate)`)]
+      }
+    } else {
+      value = theme.resolve(candidate.value.value, ['--rotate']) ?? `${candidate.value.value}deg`
```

Comment:
We should validate if `candidate.value.value` in the fallback case `${candidate.value.value}deg`. Similar to how we handle it in this pr: #13245 

Otherwise `rotate-foo` will turn into:
```css
.rotate-foo {
  rotate: foodeg;
}
```

Reply:
`rotate-potato` test added for safety. 🥔